src/components/ReservationModal.tsx

```diff
@@ -0,0 +1,93 @@
+import React, {useState} from 'react';
+import styled from 'styled-components';
+import { WHITE, BLACK } from '../colors';
+import { Modal } from 'antd';
+
+export enum ReservationModalType {
+  OPEN = 'OPEN',
+  TAKEN = 'TAKEN',
+  RESERVED = 'RESERVED'
+}
+
+interface ReservationModalProps {
+  readonly status: ReservationModalType;
+  readonly blockID: number;
+  readonly onOk: () => void;
+  readonly onCancel: () => void;
+  readonly isVisible: boolean;
+}
+
+const StyledModal = styled(Modal)`
+    height: 40px;
+    background: ${WHITE};
+    color: ${BLACK};
+    font-size: 5px;
+    font-weight: 400;
+`;
+
+const ReservationModal: React.FC<ReservationModalProps> = ({ 
+  status, blockID, onOk, onCancel, isVisible}) => {
+    const [visibility, setVisibility] = useState(isVisible);
```

Comment:
I don't think that we want to store this in local component state, we should just directly use `isVisible` prop and hold the state in the parent component.